paper.md

```diff
@@ -0,0 +1,95 @@
+---
+
+title: 'Hypothesis: A new approach to property-based testing'
+tags:
+    - Python
+    - testing
+    - test-case reduction
+    - test-case generation
+    - property-based testing
+
+authors:
+
+    - name: David R. MacIver
+      orcid: 0000-0002-8635-3223
+    - name: Zac Hatfield Dodds
+      orcid: 0000-0002-8646-8362
+    - name: many other contributors
+---
+
+# Summary
+
+*Property-based testing* is a style of testing popularised by the QuickCheck family of libraries,
+first in Haskell [@DBLP:conf/icfp/ClaessenH00] and later in Erlang [@DBLP:conf/erlang/ArtsHJW06],
+which integrates generated test cases into existing software testing workflows:
+Instead of tests which provide examples of a single concrete behaviour,
+tests specify properties that hold for a wide range of inputs,
+which the testing library then attempts to generate test cases to refute.
+For a general introduction to property-based testing, see [@PraiseOfPBT].
+
+Hypothesis is a mature and widely used^[It has over 100,000 downloads per week, over a thousand open source projects use it, and in 2018 more than 4% of Python users surveyed by the PSF reported using it.] property-based testing library for Python,
+and is one of very few widely used examples of such a library outside of functional programming languages.
+It will be of interest both to researchers using Python for developing scientific software,
+and to software testing researchers as a platform for research in its own right.
+
+# Hypothesis for Testing Scientific Software
+
+Python has a rich and thriving ecosystem of scientific software, and Hypothesis is helpful for ensuring its correctness.
+This is particularly useful for foundational libraries on which the scientific software ecosystem is built,
+but any researcher who tests their software in Python can benefit from these facilities. 
+
+For example, it has found bugs in astropy ([@astropy:2018]) and numpy ([@DBLP:journals/cse/WaltCV11]),
+both of which are foundational libraries on which a significant amount of research is based.
```

Comment:
If there are URLs to relevant illustrative bug reports it would be nice to cite them.

Reply:
- https://github.com/numpy/numpy/issues/10930, https://github.com/numpy/numpy/issues/13089, https://github.com/numpy/numpy/issues/14239
- https://github.com/astropy/astropy/pull/9328 (see comment in test function; also https://github.com/astropy/astropy/issues/9017#issuecomment-538791736)

Also exciting that the Astropy coordination meeting this December has a proposal to add Hypothesis to `pytest-astropy` and encourage its use in the ecosystem.